cms/utils/permissions.py

```diff
@@ -188,32 +188,38 @@ def get_page_actions_for_user(user, site):
     )
     nodes = [page.node for page in pages]
     pages_by_id = {}
+    page_permissions_chunks = []
+
+    for offset in range(0, pages.count(), 500):
```

Comment:
not a fan of having a magical number in here. As it only affects SQLite we should also probably target only SQLite with such an exception. I'm sadly not deep into the permission system by itself. WDYT @yakky or @czpython ?

Reply:
I will make this number more explicit by setting a constant (something like PAGE_PERMISSION_CHUNK_SIZE)
We can then have different numbers for different database backends and make it very large (to the point of being infinite) for DBMS other than sqlite